### PR TITLE
AP-45 integrate terminal

### PR DESCRIPTION
**Pull request by RobinSiep**
## Code R Ticket
_Please provide a link to the Code R Jira ticket if applicable_

<https://partypeak.atlassian.net/browse/AP-45>

## Description
Adds an icon to the agent list item to spawn a terminal. This terminal is an overlapping "window" that's draggable and resizable. It has two buttons to close the terminal window and open it in an actual new browser window / tab.

On mobile the terminal is always opened in a new browser window / tab.

image

image

## Dependencies
_Are there any dependencies in relation to pull requests in other Apollo repositories? If so please provide links to these pull requests_

* https://github.com/thecoderstudio/apollo-web/pull/5

## Additional notes
Window is draggable and all that. It's also resizable which seems to work fine front-end but on the agent side the PTY logically doesn't get informed which means the new size doesn't get reflected while reading.

### CODE REVIEW

File: src/components/agent-list/AgentListItem.jsx [hunk 1]
```diff
@@ -1,83 +1,112 @@
...
+      text-align: center;
+      width: 100%;
     `
   }
 `;
 
-const ConnectionStateContent = styled.div`
-  grid-column: connection-status;
+const TerminalIcon = styled(Icon)`
+  color: ${props => props.active ? props.theme.white : props.theme.inactive};
```
Comment (rikvanderwerf) [inline src/components/agent-list/AgentListItem.jsx:64]:
```suggestion
  color: ${props => props.active ? props.theme.white : props.theme.inactive};
  cursor: pointer;
```

**Review by rikvanderwerf — CHANGES_REQUESTED** [review]

File: src/components/agent-list/AgentListItem.jsx (hunk 1 above)
Comment (robinsiep) [reply src/components/agent-list/AgentListItem.jsx:64]: Included on `Icon`. Toggled depending on its active state.

Comment (rikvanderwerf) [reply src/components/agent-list/AgentListItem.jsx:64]: Ahh ok, but you can still click it if its disconnected. But i assume that its intended then?

Comment (robinsiep) [reply src/components/agent-list/AgentListItem.jsx:64]: Ah no in retrospect that's not what you'd expect.

**Review by rikvanderwerf — APPROVED** [review]

**Pull request merged**